pkg/updater/trainingjob_updater.go

```diff
@@ -849,8 +852,9 @@ func (j *JobUpdater) addLabelToPods(podType PodType) (bool, error) {
 			return false, err
 		}
 
-		if int32(idx) > *j.Job.Spec.Trainer.ReplicaSpec.Spec.Parallelism-1 {
-			log.Warn("Idx exceeded parallelism", "current id", idx)
+		// FIXME: the loop will now index up to desiredPodNum+1 pods instead of desiredPodNum. Is that the desired behavior?
```

Comment:
Maybe print out more details about the pod list will be good to see whether this will be triggered.